sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -190,18 +190,11 @@ def __init__(self, packages, options, environment_version, pipeline_url):
     pool = dataflow.WorkerPool(
         kind='local' if self.local else 'harness',
         packages=package_descriptors,
-        # https://issues.apache.org/jira/browse/BEAM-3116
-        # metadata=dataflow.WorkerPool.MetadataValue(),
```

Comment:
<!--thread_id:cc_149550196_t; commit:3dfcb44475c92cea85cdceebe1d0cce0ca0e31cf; resolved:1-->
<!--section:context-quote-->
> **robertwb** wrote:
> Should this line be uncommented rather than deleted?

<!--section:body-->
Actually they can't work right now. The issue is an internal Dataflow bug that manifests as [BEAM-3116](https://issues.apache.org/jira/browse/BEAM-3116). TL;DR: we cannot use worker pool metadata at all.

Template jobs are encoded in proto3 JSON format, more or less, and when they are run they are decoded via proto2 JSON format. This bug went undetected until now because there were no nonempty maps. Longer term, templates should use a proto binary encoding and perhaps be created service-side.

For now, putting this in pipeline options instead, as a generic "bag of values".

Reply:
Shouldn't we leave these comments here as a reminder that we need to fix the bug and pass the pool.metadata (here or elsewhere)? Or is this just some cleanup (that's irrelevant to the other changes in this PR?)

Reply:
I have an internal bug filed, and discussion on going, on fixing the pool metadata situation as well as designing the very best place for this sort of thing. IMO for the purposes of this PR both pool metadata and pipeline options are both sort of fine places to put this data, but neither are obviously the Right Thing To Do. So I really do mean to fully abandon ever trying to put this data in the pool metadata.

Reply:
Thanks for the context. LGTM.